encoding/protobuf/unaggregated_iterator.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package protobuf
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/m3db/m3metrics/generated/proto/metricpb"
+	"github.com/m3db/m3x/pool"
+)
+
+// UnaggregatedIterator decodes unaggregated metrics.
+type UnaggregatedIterator interface {
```

Comment:
Do you need a `Reset()` function for reusing?

Reply:
Turns out I don't reuse it at the moment. So leave it out for now, can add when needed (probably wont' need it).

Reply:
Interesting, don't you need this to decode data from collectors?

Reply:
It's one per connection, and if it encounters an error, i just close the existing one and create a new one. Since the connections are usually long-lived, the amount of new allocations is fairly low.